hal/src/gcc/core_hal.cpp

```diff
@@ -52,25 +52,32 @@ void setLoggerLevel(LoggerOutputLevel level)
     log_level = level;
 }
 
-void log_message_callback(const char *msg, int level, const char *category, uint32_t time, const char *file, int line,
-        const char *func, void *reserved)
+void log_message_callback(const char *msg, int level, const char *category, const LogAttributes *attr, void *reserved)
 {
     if (level < log_level) {
         return;
     }
     std::ostringstream strm;
     // Timestamp
-    strm << std::setw(10) << std::setfill('0') << time << ' ';
-    // Category (optional)
-    if (category && category[0]) {
+    if (attr->has_time) {
+        strm << std::setw(10) << std::setfill('0') << attr->time << ' ';
```

Comment:
we currently don't use the stream classes, since I thought they were quite heavy...do you know what overhead they bring?